Fix typo in docs (#2030)
FluxML/Flux.jl: docs/src/training/training.md

At first glance, it may seem strange that the model that we want to train is not part of the input arguments of `Flux.train!` too. However the target of the optimizer is not the model itself, but the objective function that represents the departure between modelled and observed data. In other words, the model is implicitly defined in the objective function, and there is no need to give it explicitly. Passing the objective function instead of the model and a cost function separately provides more flexibility and the possibility of optimizing the calculations.

## Model parameters

The model to be trained must have a set of tracked parameters that are used to calculate the gradients of the objective function. In the [basics](../models/basics.md) section it is explained how to create models with such parameters. The second argument of the function `Flux.train!` must be an object containing those parameters, which can be obtained from a model `m` as `Flux.params(m)`.

Such an object contains a reference to the model's parameters, not a copy, such that after their training, the model behaves according to their updated values.

> Handling all the parameters on a layer-by-layer basis is explained in the [Layer Helpers](../models/basics.md) section. Also, for freezing model parameters, see the [Advanced Usage Guide](../models/advanced.md).

Reviewer: ```suggestion
Handling all the parameters on a layer-by-layer basis is explained in the [Layer Helpers](../models/basics.md) section. For freezing model parameters, see the [Advanced Usage Guide](../models/advanced.md).
```

data = ncycle([(x, y)], 3)
```

> It's common to load the `x`s and `y`s separately. In this case, you can use `zip`:

Reviewer: ```suggestion
It's common to load the `x`s and `y`s separately. Here you can use `zip`:
```

## Custom Training loops

The `Flux.train!` function can be very convenient, especially for simple problems.

> It's also very flexible with the use of callbacks.
> But for some problems it's much cleaner to write your own custom training loop.

Reviewer: ```suggestion
For some problems, however, it's much cleaner to write your own custom training loop.
```
I don't agree with that first line :smile:

    end
    # Insert whatever code you want here that needs training_loss, e.g. logging.
    # logging_callback(training_loss)

>     # Insert whatever code you want here that needs gradient.
>     # E.g. logging with TensorBoardLogger.jl as a histogram so you can see if it is becoming huge.

Reviewer: ```suggestion
    # Insert whatever code you want here that needs gradients.
    # e.g. logging histograms with TensorBoardLogger.jl to check for exploding gradients.
```
